src/jttt/Core/Board/Board.java

```diff
@@ -13,29 +13,37 @@
 public class Board {
     private static final int POSITIVE_OFFSET = 1;
     private static final int NEGATIVE_OFFSET = -1;
+    private final DisplayStyler styler;
     private int dimension;
     private List<Mark> cells;
 
     public Board(int dimension) {
         this.dimension = dimension;
         this.cells = new ArrayList<>(generateEmptyCells());
+        this.styler = new DisplayStyler();
     }
 
     public Board(int dimension, List<Mark> initialState) {
         this.dimension = dimension;
         this.cells = initialState;
+        this.styler = new DisplayStyler();
     }
 
-    public Board(List<Mark> cells) {
-        this.dimension = (int) Math.sqrt(cells.size());
-        this.cells = cells;
+    public Board(int dimension, DisplayStyler boardStyle) {
+        this.dimension = dimension;
+        this.cells = new ArrayList<>(generateEmptyCells());
+        this.styler = boardStyle;
     }
 
     public Board playCounterInPosition(int position, Mark mark) {
         if (validPosition(position)) {
-            cells.set(position - POSITIVE_OFFSET, mark);
+            cells = newCellsWithNewMove(position - POSITIVE_OFFSET, mark);
```

Comment:
I'm not sure what you mean by implicit mutation, but I think, this functionality is unnecessary.  Rather it is unnecessary to replace 'cells' with an entirely new ArrayList.  I should just update/mutate the existing 'cells' list.  
Do you think doing: cells.set(position- POSITIVE_OFFSET, mark); in it's place is okay?